byterun/caml/mlvalues.h

```diff
@@ -25,6 +25,12 @@ typedef intnat value;
 typedef int32 opcode_t;
 typedef opcode_t * code_t;
 
+/* atomic_uintnat is declared as a struct type to prevent accidental
+   accesses that don't use the primitives of platform.h */
+typedef struct atomic_uintnat {
+  volatile uintnat val;
+} atomic_uintnat;
```

Comment:
See [here at the bottom of the page](https://software.intel.com/en-us/articles/coding-for-performance-data-alignment-and-structures) and [here](https://stackoverflow.com/questions/24102832/atomic-increment-is-flushing-other-cache-lines-code-and-results-provided) for some possible implications.

Alignment makes sure the atomic starts on a cache line -- otherwise the padding just extends from one cache line to the next, and doesn't prevent false sharing before the atomic. You don't want false sharing between atomics, and you don't really want atomics to share cache lines with other data either, since every update of an atomic forces cache line dumping by all other cores, which will hurt your cache performance for non-atomic data.